src/main/java/org/opentripplanner/middleware/triptracker/interactions/busnotifiers/UsRideGwinnettBusOpNotificationMessage.java

```diff
@@ -0,0 +1,108 @@
+
+import static org.opentripplanner.middleware.utils.DateTimeUtils.getOtpZoneId;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.getAgencyIdFromLeg;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.getRouteIdFromLeg;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.getStopIdFromPlace;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.getTripIdFromLeg;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.removeAgencyPrefix;
+
+/**
+ * Class containing the expected notify parameters. These will be converted to JSON and make up the body content of
+ * the request.
+ * <p>
+ * 'To' fields omitted as they are not needed for requests for single transit legs.
+ */
+public class UsRideGwinnettBusOpNotificationMessage {
+
+    public UsRideGwinnettBusOpNotificationMessage() {
+        // Required for JSON deserialization.
+    }
+
+    /** This is the date format required by the API. */
+    private static final DateTimeFormatter BUS_OPERATOR_NOTIFIER_API_DATE_FORMAT = DateTimeFormatter
+        .ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+        .withZone(getOtpZoneId());
+
+    /** This is the time format required by the API. */
+    public static final DateTimeFormatter BUS_OPERATOR_NOTIFIER_API_TIME_FORMAT = DateTimeFormatter
+        .ofPattern("HH:mm:ss")
+        .withZone(getOtpZoneId());
+
+    private static final Map<String, Integer> MOBILITY_CODES_LOOKUP = createMobilityCodesLookup();
+
+    /**
+     * Create a relationship between mobility modes and mobility codes.
+     */
+    private static Map<String, Integer> createMobilityCodesLookup() {
+        HashMap<String, Integer> codes = new HashMap<>();
+        codes.put("Device", 1);
+        codes.put("MScooter", 2);
+        codes.put("WChairE", 3);
+        codes.put("WChairM", 4);
+        codes.put("Some", 5);
+        codes.put("LowVision", 6);
+        codes.put("Blind", 7);
+        codes.put("Device-LowVision", 8);
+        codes.put("MScooter-LowVision", 9);
+        codes.put("WChairE-LowVision", 10);
+        codes.put("WChairM-LowVision", 11);
+        codes.put("Some-LowVision", 12);
+        codes.put("Device-Blind", 13);
+        codes.put("MScooter-Blind", 14);
+        codes.put("WChairE-Blind", 15);
+        codes.put("WChairM-Blind", 16);
+        codes.put("Some-Blind", 17);
+        return codes;
+    }
+
+    public String timestamp;
+    public String agency_id;
+    public String from_route_id;
+    public String from_trip_id;
+    public String from_stop_id;
+    public String to_stop_id;
+    public String from_arrival_time;
+    public Integer msg_type;
+    public List<Integer> mobility_codes;
+    public boolean trusted_companion;
+
+    public UsRideGwinnettBusOpNotificationMessage(Instant timestamp, TravelerPosition travelerPosition) {
+        var nextLeg = travelerPosition.nextLeg;
+        this.timestamp = BUS_OPERATOR_NOTIFIER_API_DATE_FORMAT.format(timestamp);
```

Comment:
This is incorrect. This will format a local time correctly but show that in UTC. You need to convert the local time to UTC. Here is the sample for the call I made at around 10:43 am EDT:
```
"{"timestamp":"2024-06-06T10:43:45.000Z"
```

Reply:
I think you mean convert UTC to local time? Change in: https://github.com/ibi-group/otp-middleware/pull/233/commits/bbfa26f154d08da4f36f887232a3418e70da467d

Reply:
I'd have to try out, but the timestamp should be the UTC equivalent of the time the location or request was emitted.

Reply:
I've based the update on the approach taken to get the `Leg.getScheduledStartTime()` (https://github.com/ibi-group/otp-middleware/blob/f3a0cfbdcae8d81b31cfe7239e6031b6c7f3588a/src/main/java/org/opentripplanner/middleware/otp/response/Leg.java#L76). This is used in `TripInstruction` and `UsRideGwinnettBusOpNotificationMessage` for the `from_arrival_time`.

Reply:
@br648  In the below example, with bus arrival time at 4:34 pm EDT (local time), even after your latest changes, the timestamp should have been 2024-06-05T20:20:24.000Z:

```
"{\"timestamp\":\"2024-06-05T16:20:24.000Z\",\"agency_id\":\"GCT\",\"from_route_id\":\"40\",\"from_trip_id\":\"t60E-b12F-sl4\",\"from_stop_id\":\"539\",\"to_stop_id\":\"351\",\"from_arrival_time\":\"16:34:56\",\"msg_type\":1,\"mobility_codes\":[],\"trusted_companion\":false}"
```